utils/cloudinfo/aws.go

```diff
@@ -23,7 +23,10 @@ import (
 )
 
 func onAWS() bool {
-	client := ec2metadata.New(session.New(&aws.Config{}))
+	// by default, we set retries to 0 for this operation.
+	// when not running on AWS, not setting retries can cause the operation to block
```

Comment:
I caught it running e2e not GCE or AWS.  It was very obvious.

On Wed, Mar 9, 2016 at 2:23 PM, Vish Kannan <notifications@github.com>
wrote:

> In utils/cloudinfo/aws.go
> <https://github.com/google/cadvisor/pull/1152#discussion_r55573268>:
>
> > @@ -23,7 +23,10 @@ import (
> >  )
> >
> >  func onAWS() bool {
> > -	client := ec2metadata.New(session.New(&aws.Config{}))
> > +	// by default, we set retries to 0 for this operation.
> > +	// when not running on AWS, not setting retries can cause the operation to block
>
> I wonder why the e2e tests did not catch this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/1152/files#r55573268>.
>